### PR TITLE
bug (refs T30876): missing map drawings export

### PR DESCRIPTION
**Pull request by breierDemos**
**Ticket:** https://yaits.demos-deutschland.de/T30876

Description: https://php.watch/versions/8.0/gdimage indicates that MapScreenshotter::saveImageToFile::assertResource()
needs to be adjusted. ```is_resource($resource)``` needs to be changed to ```instanceof GdImage```. Also adjusted some type hints.

### How to review/test
Not testable at the moment, because RabbitMQ is currently not working.
Therefore it could only be tested on dev at the moment.

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly

### CODE REVIEW

**Comment by breierDemos** [comment]
>
Adjusted in [e6b6c31](https://github.com/demos-europe/demosplan-core/pull/637/commits/e6b6c315297fe363fbb1ca602c59b6caf46b34e5)

**Comment by breierDemos** [comment]
> This doesn't seem right. The docblock type-hints imply that `GdImage` is the only valid input. In this case we can move that type into the signature, which makes the `assertGdImage` obsolete. It seems we need to move the `assertGdImage` somewhere further up to the first place we want to be sure we're handling a `GdImage` instead of a `resource`.

Tried to address your concerns in [e6b6c31](https://github.com/demos-europe/demosplan-core/pull/637/commits/e6b6c315297fe363fbb1ca602c59b6caf46b34e5)

File: demosplan/DemosPlanMapBundle/Utilities/MapScreenshotter.php [hunk 1]
```diff
@@ -580,18 +570,6 @@ protected function adjustPictureSize(float $top, stdClass $viewport, float $bott
...
-     */
-    public function assertResource($resource): bool
-    {
-        if (false === is_resource($resource)) {
-            throw new InvalidArgumentException(sprintf('Argument must be a valid resource type. %s given.', gettype($resource)));
-        }
-
-        return true;
-    }
-
```
Comment (dresslerdemos) [inline demosplan/DemosPlanMapBundle/Utilities/MapScreenshotter.php:594]: Still used in unit tests.

**Review by dresslerdemos — CHANGES_REQUESTED** [review]
Adjusted types in https://github.com/demos-europe/demosplan-core/pull/637/commits/3d0967e62e505cdfe37ada37a0e4283788446802

File: demosplan/DemosPlanMapBundle/Utilities/MapScreenshotter.php (hunk 1 above)
Comment (breierDemos) [reply demosplan/DemosPlanMapBundle/Utilities/MapScreenshotter.php:594]: Adjusted in [51892aa](https://github.com/demos-europe/demosplan-core/pull/637/commits/51892aaf334ff1eee681ceb591bc9342ff3b3bdb)

**Review by dresslerdemos — APPROVED** [review]
Minor adjustment to `assertGdImage` (c422b98d3a3c9381d86e5acf289108d9d1361adf).

**Pull request merged**